projects/storefrontlib/src/cms-components/cart/cart-details/cart-details.component.ts

```diff
@@ -26,8 +24,22 @@ export class CartDetailsComponent implements OnInit {
   }
 
   getAllPromotionsForCart(cart: Cart): Cart[] {
-    const potentialPromotions = cart.potentialOrderPromotions || [];
-    const appliedPromotions = cart.appliedOrderPromotions || [];
+    const potentialPromotions = [];
+    if (cart.potentialOrderPromotions) {
```

Comment:
IF statements are not needed here as push with `...` handles the case

Reply:
These IF statements seem required.
(I really was hoping for a cleaner solution haha)

If we do `[...[]]` then it will not push anything since the array is empty.
However, if we do `[...undefined]` it will throw an error. 
It's possible that the results are undefined.

Let me know what you think

Reply:
Oh I didn't consider that. What about this?
`const potentialPromotions = [
  ...(cart.potentialOrderPromotions || []),
  ...(cart.potentialProductPromotions || []),
];`

`    const appliedPromotions = [
      ...(cart.appliedOrderPromotions || []),
      ...(cart.appliedProductPromotions || []),
    ];`

Reply:
Works, clever solutions